extensions/amp-consent/0.1/test/test-consent-state-manager.js

```diff
@@ -264,6 +258,19 @@ describes.realWin('ConsentStateManager', {amp: 1}, (env) => {
         );
       });
 
+      it('should call purpose consent resolver on update', () => {
+        const resolverSpy = env.sandbox.spy(manager, 'hasAllPurposeConsents');
```

Comment:
Spy on the prototype?

Reply:
What does spying on the prototype achieve here?

Reply:
I'm not sure. I generalized your comment of spying on the prototype as something that should always be done.

Reply:
Oh no, I meant that in my experience, spying on the prototype can be helpful in some situations, but I think in general spying on the object is better.